crates/turbopack-ecmascript/src/chunk/optimize.rs

```diff
@@ -269,11 +272,69 @@ async fn merge_duplicated_and_contained(
     Ok(())
 }
 
-async fn merge_by_size(chunks: &[EcmascriptChunkVc]) -> Result<Vec<EcmascriptChunkVc>> {
+async fn merge_to_limit(
+    chunks: Vec<(EcmascriptChunkVc, Option<ChunksVc>)>,
+    target_count: usize,
+) -> Result<Vec<EcmascriptChunkVc>> {
+    let mut remaining = chunks.len();
+    let mut chunks = chunks.into_iter().enumerate().collect::<Vec<_>>();
```

Comment:
I think the enumerate (and the nested tuple complications below) is unnecessary? I assume you're doing it to ensure a stable sorting of the chunks, but Rust's sort it stable already.